extension/chrome/settings/modules/keyserver.ts

```diff
@@ -96,7 +98,14 @@ View.run(class KeyserverView extends View {
       return;
     }
     try {
-      await this.pubLookup.attester.initialLegacySubmit(String($(target).attr('email')), mostUsefulPrv.keyInfo.public);
+      const email = String($(target).attr('email'));
+      // Use submitPrimaryEmailPubkey if email is primary email
+      if (email === this.acctEmail) {
+        const idToken = await InMemoryStore.get(this.acctEmail, InMemoryStoreKeys.ID_TOKEN);
```

Comment:
There is a method in GoogleAuth that always gives you fresh id token if needed. This one may be stale.

Reply:
Hmm.. Which method do you mean in GoogleAuth?

Reply:
You're right, there isn't any. And the id token is kept reasonably fresh through other mechanism (when Google API is called and when access token is refreshed). The browser extension is not designed ideally there, but it does work, and you've done it correctly.